elide-standalone/src/main/java/com/yahoo/elide/standalone/config/ElideStandaloneSettings.java

```diff
@@ -167,6 +169,60 @@ default boolean enableJSONAPI() {
     default boolean enableGraphQL() {
         return true;
     }
+    
+    /**
+     * Enable the support for Async querying feature. If false, the async feature will be disabled.
+     *
+     * @return Default: False
+     */
+    default boolean enableAsync() {
+        return false;
+    }
+
+    /**
+     * Enable the support for cleaning up Async query history. If false, the async cleanup feature will be disabled.
+     *
+     * @return Default: False
+     */
+    default boolean enableAsyncCleanup() {
+        return false;
+    }
+
+    /**
+     * Thread Size for Async queries to run in parallel.
+     *
+     * @return Default: 5
+     */
+    default Integer getAsyncThreadSize() {
+        return 5;
+    }
+
+    /**
+     * Maximum Query Run time for Async Queries to mark as TIMEDOUT.
+     *
+     * @return Default: 60
+     */
+    default Integer getMaxRunTimeMinutes() {
+        return 60;
+    }
+
+    /**
+     * Number of days history to retain for async query executions and results.
+     *
+     * @return Default: 7
+     */
+    default Integer getQueryCleanupDays() {
+        return 7;
+    }
+
+    /**
+     * Implementation of AsyncQueryDAO to use.
+     *
+     * @return AsyncQueryDAO type object.
+     */
+    default AsyncQueryDAO getAsyncQueryDAO() {
```

Comment:
Instead of resolving this as a setting - this can be injected into the async code (either via lifecycle hook or bean).

Reply:
As discussed will leave it as is.